doc/main/reference/try_put_and_wait.rst

```diff
@@ -0,0 +1,322 @@
+.. _try_put_and_wait:
+
+Waiting for Single Messages in Flow Graph
+=========================================
+
+.. contents::
+    :local:
+    :depth: 1
+
+Description
+***********
+
+This feature adds a new ``try_put_and_wait`` interface to the receiving nodes in the Flow Graph.
+This function allows to wait until completion of single message, submitted to ``try_put_and_wait`` as an input by the Flow Graph in
+opposite to ``graph.wait_for_all()`` function that waits for completion of all tasks in the graph.
+
+``node.try_put_and_wait(msg)`` performs ``node.try_put(msg)`` on the node and waits until the work on ``msg`` is completed.
+Therefore, the following conditions are true:
+
+* Any task initiated by any node in the Flow Graph that involves on working with ``msg`` or any other intermediate result
+computed from ``msg`` is completed.
+* Any intermediate result computed from ``msg`` is retrieved from any buffer in the graph.
+
+.. caution::
+
+    To prevent ``try_put_and_wait`` calls from infinite blocking, avoid using buffering nodes at the end of the Flow Graph since the final result
```

Comment:
```suggestion
    To prevent ``try_put_and_wait`` calls from infinite waiting, avoid using buffering nodes at the end of the Flow Graph since the final result
```